fix(#1803): fix do/don't grids and component content grid

bdfranck opened the pull request:
This PR addresses the issues identified in https://github.com/GovAlta/ui-components/issues/1803. It makes the following changes:

- Reduces the minimum grid width of the "Do &  Don't" columns from `50ch` to `36ch`
- Swiches the `ComponentContent` grid to a single column when the table of contents disappears below `1230px` width.
- Adds a `min-width` of 0 to `.component-content--content`. This prevents wider content from [growing the column width past the available space](https://css-tricks.com/preventing-a-grid-blowout/).
- Fixes an `undefined` class name in `ComponentContent`

I determined that `36ch` is the maximum width at `642px` without overflowing the content area.

<img width="653" alt="image" src="https://github.com/GovAlta/ui-components-docs/assets/1479091/f75a1fec-8fef-4bda-8614-1180f6e77360">

The header-related issues are captured in this other ticket: https://github.com/GovAlta/ui-components/issues/1438

Checklist:
- [x] I have run a build locally.
- [x] I have tested the functionality.

bdfranck commented:
Here's how it looks in `Alpha`...
<img width="709" alt="image" src="https://github.com/GovAlta/ui-components-docs/assets/1479091/b25eca08-9920-460c-a8fd-05327d7f5e11">


And here's how it looks in this PR:
<img width="717" alt="image" src="https://github.com/GovAlta/ui-components-docs/assets/1479091/c0285006-9672-4e5c-b251-e9239d75c856">

bdfranck commented:
@twjeffery spotted an additional issue that is present on `alpha`: the content doesn't fill the full width when <624px wide.

![image](https://github.com/GovAlta/ui-components-docs/assets/1479091/9acae79f-8d4f-4c25-87bc-678375d4b29c)

Here's how it looks on the production site:
![image](https://github.com/GovAlta/ui-components-docs/assets/1479091/8f3db1cc-08f2-42be-873d-6afc98232eab)

bdfranck commented:
I'm exploring what the above issues is...

bdfranck commented:
@twjeffery also noticed that the table on the Error Text page was increasing the width of `component-content--content`. It looks like the table's horizontal scrolling isn't working. 

![image](https://github.com/GovAlta/ui-components-docs/assets/1479091/90bcba23-58e6-4b67-a1e8-e0bc3675df9e)

I made the following updates to this PR to address the issue:
- Add `min-width: 0` to `component-content--content` so the column width stays within the container
- Wrap the table with `<div style={{position: "relative", overflowX: "auto"}}>` so it scrolls

I don't think the wrapper `div` is a great solution so I'm open to suggestions.

chrisolsen commented on src/components/component-content/component-content.css:
```diff
@@ -47,9 +47,14 @@ h3[id^="toc"] + p {
     display: grid;
     grid-template-columns: 1fr 12rem;
     grid-gap: var(--goa-space-2xl);
+
+    @media screen and (max-width: 1230px) {
+        grid-template-columns: 1fr;
+      }
```
Does this work? Nesting css is now possible, but last I checked it didn't yet work with the svelte compiler.

If it does work, then just a small fix to the indentation is needed.

bdfranck replied:
Yup! It works without issues, and we already use it in the ToC component:

https://github.com/GovAlta/ui-components-docs/blob/988a16991bd86e4a7db9daf13332616353b272d9/src/components/table-of-contents/toc.module.css#L1-L11

chrisolsen commented on src/components/component-content/component-content.css:
```diff
@@ -47,9 +47,14 @@ h3[id^="toc"] + p {
 .component-content--content {
     flex: 1 1 auto;
     container: inline-size;
+    min-width: 0;
```
Is this needed?

bdfranck replied:
Yes, it prevents wide content in `.component-content--content` from [blowing out of the grid column](https://css-tricks.com/preventing-a-grid-blowout/). Mind you, that CSS Tricks article is old so there may be a better method that I'm not aware of.

chrisolsen replied:
I figured this was some sort of css hack :)

chrisolsen commented on src/routes/content/HelperText.tsx:
```diff
@@ -46,128 +48,130 @@ export default function HelperTextPage() {
-          </tr>
-        </tbody>
-      </GoATable>
+      <div style={{position: "relative", overflowX: "auto"}}> 
```
What is the reason for this? Previously there was an issue with the GoATable with horizontal scrolling, but that was recently fixed, so maybe this is no longer necessary.

bdfranck replied:
Good to know! I'll pull this out.

bdfranck commented:
@chrisolsen Thanks for the feedback! I've updated my PR with the following changes:

- Removed the scrolling wrapper div from the `HelperText` table
- Reduced the indentation in `component-content.css` from 4 to 2 spaces
